pkg/client/src/app/pages/applications/analysis-wizard/components/add-custom-rules.tsx

```diff
@@ -116,11 +116,11 @@ export const AddCustomRules: React.FunctionComponent<IAddCustomRules> = ({
           );
           handleReadFail(error, 100, file);
         } else {
-          if (validateXMLFile(data as string))
-            handleReadSuccess(data as string, file);
+          const validatedXML = validateXMLFile(data as string);
+          if (validatedXML === "") handleReadSuccess(data as string, file);
```

Comment:
IMO, This is tough to follow. Can we add an area to store the error and use that to conditionally validate the read success? Just something a little more clear would be nice. 

For example, I think it would be helpful to return an object that has the error message to pull off of if it exists & any other relevant validation info rather than returning a string.

Reply:
Good idea, thanks, done.